.ci/Jenkinsfile

```
@@ -413,85 +417,87 @@ def archiveArtifactsSafe(remotePath, artifacts) {
 }
 
 def packageStoragePublish() {
-  stage('Publish to Package Storage v2') {
-    if (env.BRANCH_NAME != 'main' && !env.BRANCH_NAME.startsWith('backport-')) {
-      echo "packageStoragePublish: not the main branch or a backport branch, nothing will be published"
-      return
-    }
+  if (env.BRANCH_NAME != 'main' && !env.BRANCH_NAME.startsWith('backport-')) {
+    echo "packageStoragePublish: not the main branch or a backport branch, nothing will be published"
+    return
+  }
 
-    deleteDir()
-    unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
-    useElasticPackage()
-
-    // Build only unpublished files (need to read version from manifest.yml)
-    def unpublishedPresent = false
-    dir("${BASE_DIR}/packages") {
-      findFiles()?.findAll{ !it.name.endsWith('@tmp') }?.collect{ it.name }?.sort()?.each {
-        dir("${it}") {
-          def manifest = readYaml(file: "manifest.yml")
-          if (manifest != null && manifest["version"] != null) {
-            def packageZip = it + "-" + manifest["version"] + ".zip"
-            if (isAlreadyPublished(packageZip)) {
-              return
-            }
+  deleteDir()
+  unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
+  useElasticPackage()
+
+  // Build only unpublished files (need to read version from manifest.yml)
+  def unpublishedPresent = false
+  dir("${BASE_DIR}/packages") {
+    findFiles()?.findAll{ !it.name.endsWith('@tmp') }?.collect{ it.name }?.sort()?.each {
+      dir("${it}") {
+        def manifest = readYaml(file: "manifest.yml")
+        if (manifest != null && manifest["version"] != null) {
+          def packageZip = it + "-" + manifest["version"] + ".zip"
+          if (isAlreadyPublished(packageZip)) {
+            return
           }
+        }
 
-          withEnv(["PATH=${env.WORKSPACE}/${BASE_DIR}/build:${env.PATH}"]) {
+        withEnv(["PATH=${env.WORKSPACE}/${BASE_DIR}/build:${env.PATH}"]) {
+          try {
             sh(label: "Build integration as zip: ${it}", script: "elastic-package build --zip")
+          }catch(error){
+            echo("Build paackage for ${it} failed")
```

```suggestion
            echo("Build package for ${it} failed")
```
If this job is only run on merges on main and backport branches, maybe we can still mark the build as failed or unstable, without interrupting it.

---

Updated. 
I see that it could be used `catchError` to change the stage result